subprojects/core-api/src/main/java/org/gradle/api/artifacts/ArtifactCollection.java

```diff
@@ -43,6 +45,16 @@
      */
     Set<ResolvedArtifactResult> getArtifacts();
 
+    /**
+     * Returns the resolved artifacts as a {@link Provider} of {@link ResolvedArtifactResult} instances.
+     * The returned {@link Provider} is live, and tracks the producer tasks of this artifact collection.
+     * The provider will resolve the artifact metadata and download the artifact files as required.
+     *
+     * @since 7.4
+     */
+    @Incubating
+    Provider<Set<ResolvedArtifactResult>> getResolvedArtifacts();
```

Comment:
Do you see this as becoming the only option in the future?

Reply:
I would say yes